urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/utilities/PaintUtilities.java

```diff
@@ -0,0 +1,110 @@
+package permlab.utilities;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import permlib.Permutation;
+import permlib.utilities.RestrictedPermutation;
+
+/**
+ * Provides paint methods for permutations and restricted permutations. Having this
+ * in a separate class avoid code duplication when exporting to image formats.
+ * 
+ * @author M Belton
+ */
+public class PaintUtilities {
+
+    public static final Color RULE_RESTRICTED = new Color(50, 50, 50, 150);
+    public static final Color USER_RESTRICTED = new Color(120, 120, 120, 100);
+    public static final int DEFAULT_POINT_SIZE = 20;
+
+    public static void paint(Graphics g, RestrictedPermutation restPerm, int width, int height, boolean monotoneConstraints, boolean showGrid) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.